wordpress.org/public_html/wp-content/plugins/wporg-gp-virtual-projects/wporg-gp-virtual-projects.php

```diff
@@ -0,0 +1,181 @@
+<?php
+/**
+ * Plugin Name: WPORG Virtual projects for patterns.
+ * Description: Creates virtual projects for the patterns.
+ * Author: the WordPress team.
+ */
+class WPorg_Virtual_Projects {
+
+	/**
+	 * Adds filters to subprojects and to project status.
+	 */
+	public function __construct() {
+		add_filter( 'gp_subprojects', array( $this, 'add_virtual_projects' ), 10, 1 );
+		add_filter( 'gp_get_project_status', array( $this, 'gp_get_project_status' ), 10, 4 );
+	}
+
+	/**
+	 * Adds virtual projects for the patterns, so each pattern is shown as a different project.
+	 *
+	 * The received and returned data is an array with:
+	 * - $data['project']       The main project.
+	 * - $data['sub_projects']  The subprojects.
+	 * - $data['pages']         Pagination information.
+	 *
+	 * @param array $data The project, subprojects and pages.
+	 *
+	 * @return array The project, subprojects (or virtual projects) and pages.
+	 */
+	public function add_virtual_projects( $data ) {
+		global $wpdb;
+		if ( 'Patterns' != $data['project']->name ) {
+			return $data;
+		}
+
+		$url_projects      = array();
+		$pattern_originals = $wpdb->get_results(
+			"SELECT `references` 
+			FROM {$wpdb->gp_originals}  
+			WHERE `project_id` = 473698 
+				AND `status` = '+active'"
+		);
+		foreach ( $pattern_originals as $row ) {
+			$row->references = trim( preg_replace( '/\s+/', ' ', $row->references ) );
+			$urls            = explode( ' ', $row->references );
+			foreach ( $urls as $url ) {
+				$url_projects[] = $url;
+			}
+		}
+		$url_projects = array_unique( $url_projects );
+		sort( $url_projects );
+
+		$pages['pages'] = (int) ceil( count( $url_projects ) / 21 );
+		if ( null == $data['pages']['page'] ) {
+			$pages['page'] = 1;
+		} else {
+			$pages['page'] = $data['pages']['page'];
+		}
+		if ( null == $data['pages']['per_page'] ) {
+			$pages['per_page'] = 21;
+		} else {
+			$pages['per_page'] = $data['pages']['per_page'];
+		}
+		$pages['results'] = count( $url_projects );
+
+		$url_projects = array_slice( $url_projects, ( $pages['page'] - 1 ) * $pages['per_page'], $pages['per_page'], true );
+
+		$url_prefix = 'https://wordpress.org/patterns/pattern/';
+
+		foreach ( $url_projects as $row ) {
+			if ( substr( $row, 0, strlen( $url_prefix ) ) == $url_prefix ) {
+				$name = substr( $row, strlen( $url_prefix ) );
+			} else {
+				$name = $row;
+			}
+			$subproject                      = new GP_Project();
+			$name                            = rtrim( $name, '/' );
+			$subproject->slug                = $name;
+			$subproject->id                  = $name;
+			$name                            = ucfirst( str_replace( '-', ' ', $name ) );
+			$subproject->name                = $name;
+			$subproject->description         = $name;
+			$subproject->path                = 'patterns/' . $subproject->slug;
```

Comment:
This leads to a 404. The idea was to directly link to a reference search like this:
```suggestion
			$subproject->path                = '/../../../projects/patterns/core/' . $locale_slug . '/' . $set_slug. '/?filters[term_scope]=scope_references&filters[term]=' . urlencode(    $url ) . '&';
```